presto-main/src/main/java/com/facebook/presto/type/DecimalCasts.java

```diff
@@ -418,15 +414,15 @@ public static long shortDecimalToReal(long decimal, long precision, long scale,
     public static long longDecimalToReal(Slice decimal, long precision, long scale, BigInteger tenToScale)
     {
         BigInteger decimalBigInteger = decodeUnscaledValue(decimal);
-        BigDecimal bigDecimal = new BigDecimal(decimalBigInteger, (int) scale);
+        BigDecimal bigDecimal = new BigDecimal(decimalBigInteger, intScale(scale));
         return floatToRawIntBits(bigDecimal.floatValue());
     }
 
     @UsedByGeneratedCode
     public static long doubleToShortDecimal(double value, long precision, long scale, long tenToScale)
     {
         BigDecimal decimal = doubleToBigDecimal(value, precision, scale);
-        decimal = decimal.setScale((int) scale, ROUND_HALF_UP);
+        decimal = decimal.setScale(intScale(scale), HALF_UP);
```

Comment:
I see you are consolidating everything to `HALF_UP` vs `ROUND_HALF_UP`.  How did you make this decision?